Add support for time_end in NaiveCertificateManager

kelunik opened the pull request:
I can also add support for `time_start_and_duration`, but I'm not sure which unit to use, as the unit for `Duration` can be years.

riebl commented:
I see no problem to use `vanetza::common::Clock` which stores microseconds in a 64bit integer (could enough for > 200.000 years). A conversion method from `security::Duration` to `std::chrono::duration` should be easy to implement.
Thanks for this pull request!

kelunik commented:
I simply went with `Time64` now, which is already defined inside vanetza.

riebl commented on vanetza/security/validity_restriction.hpp:
```diff
@@ -35,6 +36,47 @@ class Duration
+    /**
+     * Returns the duration in seconds.
+     */
+    Time64 to_seconds() const
```
This is not possible: `Time64` encodes microseconds (see section 4.2.15 of TS 103 097).
Exactly because such conversion errors happen too easily with plain integers I would favor a solution based on `vanetza::Clock::duration`.

riebl commented on vanetza/security/validity_restriction.hpp:
```diff
@@ -35,6 +36,47 @@ class Duration
+                // undefined, let's interpret it as a duration of 0, which will fail during validation
+                return 0;
+        }
+    }
```
Larger code segments should go to the source file, i.e. everything except simple getter/setter and template code.

riebl commented on vanetza/security/validity_restriction.hpp:
```diff
@@ -35,6 +36,47 @@ class Duration
+
+    uint16_t value() const
+    {
+        return m_raw & 0xCFFFF; // mask upper 3 bit
```
mask has to be 0xCFFF

kelunik commented:
I've fixed the mentioned points and added a few unit tests.

Currently `make` seems to recompile all tests every time, is there any way to only recompile the changed files and the dependent files instead?

riebl commented:
Is it really recompiling all tests? On my system it does only re-link those unit tests depending on the security module (e.g. the geonet tests).

kelunik commented:
Seems like I just got the impression because I changed the mask in `validity_restriction.hpp` which compiled a very large set of tests.

riebl commented:
Merged 0a87827bc7ce019955fb45e7f9e151b9d9ab2599

kelunik commented:
Does `to_seconds` still make sense with `std::chrono::duration`?

riebl commented:
`security::Duration::to_seconds` returns `std::chrono::seconds`, so this makes sense to me. If it were returning `Clock::duration` I would probably change it to a cast operator. Since there are no rounding errors when converting `security::Duration` to `Clock::duration` this does not introduce problems from my point of view.
